src/engraving/playback/playbackmodel.cpp

```diff
@@ -87,31 +94,84 @@ void PlaybackModel::load(Ms::Score* score, async::Channel<int, int, int, int> no
                                                        const int staffIdxFrom, const int staffIdxTo) {
         clearExpiredEvents();
         clearExpiredContexts();
-        update(tickFrom, tickTo, Ms::staff2track(staffIdxFrom, 0), Ms::staff2track(staffIdxTo, Ms::VOICES));
+
+        TrackChangesMap trackChanges;
+        update(tickFrom, tickTo, Ms::staff2track(staffIdxFrom, 0), Ms::staff2track(staffIdxTo, Ms::VOICES), &trackChanges);
+        notifyAboutChanges(std::move(trackChanges));
     });
 
     update(0, score->lastMeasure()->endTick().ticks(), 0, m_score->ntracks());
 }
 
-const PlaybackEventsMap& PlaybackModel::events(const ID& partId, const std::string& instrumentId) const
+const PlaybackData& PlaybackModel::trackPlaybackData(const ID& partId, const std::string& instrumentId) const
 {
     return m_events.at(idKey(partId, instrumentId));
 }
 
-void PlaybackModel::update(const int tickFrom, const int tickTo, const int trackFrom, const int trackTo)
+const PlaybackData& PlaybackModel::metronomePlaybackData() const
+{
+    return m_events.at(METRONOME_TRACK_ID);
+}
+
+void PlaybackModel::triggerEventsForItem(const Ms::EngravingItem* item)
+{
+    IF_ASSERT_FAILED(item) {
+        return;
+    }
+
+    if (!item->isPlayable()) {
+        return;
+    }
+
+    TrackIdKey trackId = idKey(item);
+
+    auto trackPlaybackData = m_events.find(trackId);
+    if (trackPlaybackData == m_events.cend()) {
+        return;
+    }
+
+    timestamp_t timestamp = timestampFromTicks(item->score(), item->tick().ticks());
+
+    const PlaybackEventList& eventsFromTick = trackPlaybackData->second.originData.at(timestamp);
+
+    PlaybackEventsMap result;
+
+    if (item->isChord()) {
+        for (const Ms::Note* note : Ms::toChord(item)->notes()) {
+            findEventsForNote(note, eventsFromTick, result[timestamp]);
+        }
+    } else {
+        findEventsForNote(Ms::toNote(item), eventsFromTick, result[timestamp]);
```

Comment:
I would suggest adding a check here to see if the item is really a note

Reply:
item->isPlayable() from above will ensure this